pkg/metainfo/kvmetainfo/objects.go

```diff
@@ -62,7 +77,47 @@ func (db *DB) GetObjectStream(ctx context.Context, bucket string, path storj.Pat
 // CreateObject creates an uploading object and returns an interface for uploading Object information
 func (db *DB) CreateObject(ctx context.Context, bucket string, path storj.Path, createInfo *storj.CreateObject) (object storj.MutableObject, err error) {
 	defer mon.Task()(&ctx)(&err)
-	return nil, errors.New("not implemented")
+
+	_, err = db.GetBucket(ctx, bucket)
+	if err != nil {
+		return nil, err
+	}
+
+	if path == "" {
+		return nil, storj.ErrNoPath.New("")
+	}
+
+	info := storj.Object{
+		Bucket: bucket,
+		Path:   path,
+	}
+
+	if createInfo != nil {
+		info.Metadata = createInfo.Metadata
+		info.ContentType = createInfo.ContentType
+		info.Expires = createInfo.Expires
+		info.RedundancyScheme = createInfo.RedundancyScheme
+		info.EncryptionScheme = createInfo.EncryptionScheme
+	}
+
+	// TODO: autodetect content type from the path extension
+	// if info.ContentType == "" {}
+
+	if info.RedundancyScheme == (storj.RedundancyScheme{}) {
```

Comment:
Let's add a method `IsAutomatic` or `IsZero` or `IsEmpty`. This part is bit hard to read.

Reply:
Yes, good idea.